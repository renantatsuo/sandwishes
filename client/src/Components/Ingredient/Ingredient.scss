.ingredient {
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 60px;
  width: 540px;
  padding: 10px;
  margin-bottom: 30px;

  background: #ffffff;
  border: none;
  box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.05);

  .ingredient__photo img {
    height: 40px;
    width: auto;
  }

  .ingredient__info {
    flex: 1;
    margin: 0 20px;
    text-align: left;
    font-size: 14px;

    > div {
      margin: 5px;
    }

    .ingredient__info--title {
      font-weight: bold;
    }

    .ingredient__info--desc {
      color: #999999;
    }
  }

  .ingredient__cart {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100px;

    button {
      width: 24px;
      height: 24px;
      background: transparent;
      padding: 0;

      border: none;
      cursor: pointer;

      svg {
        width: 24px;
        height: 24px;
        fill: #e73737;
      }

      &:disabled {
        cursor: default;

        svg {
          width: 24px;
          height: 24px;
          fill: #999999;
        }
      }
    }
  }
}
