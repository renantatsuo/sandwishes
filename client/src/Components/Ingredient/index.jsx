import { Add, Remove } from '@material-ui/icons'
import React, { useEffect, useState } from 'react'
import { convertFloatToMoney } from '../../Lib/Pricing'
import './Ingredient.scss'

export default function Ingredient({
  addIngredient,
  id,
  photo,
  price,
  removeIngredient,
  quantity,
  title,
}) {
  const [counter, setCounter] = useState(0)

  const handleAdd = () => {
    addIngredient(id)
    setCounter(counter + 1)
  }
  const handleRemove = () => {
    removeIngredient(id)
    setCounter(counter - 1)
  }

  useEffect(() => {
    setCounter(quantity)
  }, [quantity])

  return (
    <div className="ingredient" role="listitem">
      <div className="ingredient__photo">
        <img src={photo} alt={title} />
      </div>
      <div className="ingredient__info">
        <div className="ingredient__info--title">{title}</div>
        <div className="ingredient__info--price">
          R$ {convertFloatToMoney(price)}
        </div>
      </div>
      <div className="ingredient__cart">
        <button type="button" onClick={handleRemove} disabled={counter === 0}>
          <Remove />
        </button>
        {counter}
        <button type="button" onClick={handleAdd}>
          <Add />
        </button>
      </div>
    </div>
  )
}
