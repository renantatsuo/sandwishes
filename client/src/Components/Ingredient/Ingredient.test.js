import React from 'react'
import { render } from '@testing-library/react'
import Ingredient from './index'

describe('Test Ingredient component', () => {
  it('Should render with no errors', () => {
    const { container } = render(<Ingredient />)

    expect(
      container.querySelector('.ingredient__info--price').textContent
    ).toMatch('R$ 0,00')
    expect(container).toMatchSnapshot()
  })
})
