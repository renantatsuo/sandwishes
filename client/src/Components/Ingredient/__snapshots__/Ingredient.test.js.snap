// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Ingredient component Should render with no errors 1`] = `
<div>
  <div
    class="ingredient"
  >
    <div
      class="ingredient__photo"
    >
      <img />
    </div>
    <div
      class="ingredient__info"
    >
      <div
        class="ingredient__info--title"
      />
      <div
        class="ingredient__info--price"
      >
        R$ 
        0,00
      </div>
    </div>
    <div
      class="ingredient__cart"
    >
      <button
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13H5v-2h14v2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </button>
      <button
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
